lib/puppet/provider/pulp_schedule/api.rb

```diff
@@ -30,50 +30,42 @@ def initialize(resource={})
   end
 
   def self.get_resource_properties(repo_id)
-    all = []
-    sync_id = nil
     # get_repo_syncs return an array with all schedules and the repo type
     schedules =  @pulp.get_repo_syncs(repo_id)
-    return all unless schedules[0]
+    return [] unless schedules
+
     all_schedules = schedules[0]
     $schedules_info['repo_type'][repo_id] = schedules[1]
 
     # first get the lowest sync id
-    all_schedules.each { |schedule|
-      if sync_id && sync_id < schedule['_id']
-        sync_id = schedule['_id']
-      elsif !sync_id
-        sync_id = schedule['_id']
-      end
-    }
+    sync_id = all_schedules.map { |schedule| schedule['_id'] }.min
 
     # create resources
-    all_schedules.each { |schedule|
-      hash = {}
-      hash[:ensure] = :present
-      hash[:provider] = :pulp_schedule
-
-      # in case where we are the previously saved sync_id, we set this as the _main_ resource
+    all_schedules.map { |schedule|
       if sync_id == schedule['_id']
-        hash[:name] = repo_id
+        # in case where we are the previously saved sync_id, we set this as the _main_ resource
+        name = repo_id
         $schedules_info['repo_id'][repo_id] = schedule['_id']
       else
         # otherwise we create a dummy resource that can be purged if not needed
-        hash[:name] = schedule['_id']
+        name = schedule['_id']
         $schedules_info['schedule_id'][schedule['_id']] = repo_id
       end
 
-      hash[:schedule_time] = schedule['schedule']
-      hash[:enabled] = schedule['enabled'] == true ? :true : :false
-      hash[:failure_threshold] = schedule['failure_threshold']
-      all << new(hash)
+      new({
+        :ensure => :present,
+        :provider => :pulp_schedule,
```

Comment:
I will admit that I only copied it from the old code and I'm a bit unsure of what it exactly does.